src/contexts/AuthUIContextProvider/types.ts

```diff
@@ -15,6 +15,15 @@ type CustomRegistrationForm = {
     component: ComponentType<AccountDetailsFormProps>;
 };
 
+type InputControl = {
+    maxLength?: number;
+};
+
+type RegistrationConfig = {
+    firstName?: InputControl;
+    lastName?: InputControl;
+};
```

Comment:
I would consider not using a shared type for these if there might be things that could be different for particular fields.
```
firstName?: {maxLength?: number};
LastName?: {maxLength?: number};
```

Or maybe be a little more specific in the name of the type (InputControl feels a little general):
```
type TextFieldConfig = {
    maxLength?: number;
}
```
and then we could have different types/configuration options for different kinds of inputs (select, textfield, radio, password, etc.)